### PR TITLE
FIT image support

### PR DESCRIPTION
**Pull request by ichergui**
* The tests were done successfully when using all the machine `-cot`
* The tests were done with signed and unsigned FIT image, all devices boot successfully.

Please the documentation [here](https://github.com/OE4T/tegra-demo-distro/wiki/Tegra-&-Uboot-Flattened-Image-Tree-(FIT-image)) to get more details.

### CODE REVIEW

File: layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+DESCRIPTION = "Install tegra fitImage in the location expected by u-boot"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+COMPATIBLE_MACHINE = "(cot)"
+
+do_install() {
+    install -d ${D}/boot
+    install -m 0644 ${DEPLOY_DIR_IMAGE}/fitImage ${D}/boot/fitImage-${MACHINE}
```
Comment (ichergui) [inline layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:9]: @madisongh 
* I want to have the following lines instead 
```
root@jetson-tx2-devkit-cot:~# ls -lrta /boot/
total 13892
-rw-r--r--    1 root     root      14213662 Mar  9  2018 fitImage-4.9.201-l4t-r32.5+ge443483e8c9c
lrwxrwxrwx    1 root     root            40 Mar  9  2018 fitImage -> fitImage-4.9.201-l4t-r32.5+ge443483e8c9c
```
* With the actual implementation, I have the following lines:
```
root@jetson-tx2-devkit-cot:~# ls -lrta /boot/
total 35644
-rw-r--r--    1 root     root      39309146 Mar  9  2018 fitImage-jetson-tx2-devkit-cot
lrwxrwxrwx    1 root     root            30 Mar  9  2018 fitImage -> fitImage-jetson-tx2-devkit-cot
drwxr-xr-x    2 root     root          4096 Mar  9  2018 .
```

Comment (madisongh) [reply layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:9]: Something like this:
```
    fitImgPath=$(readlink -f ${DEPLOY_DIR_IMAGE)/fitImage)
    fitImgName=$(basename "$fitImgPath")
    install -m 0644 $fitImgPath ${D}/boot/
    ln -sf $fitImgName ${D}/boot/fitImage
```

Comment (ichergui) [reply layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:9]: Thanks @madisongh 
I will test it now and will let you know.

Comment (ichergui) [reply layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:9]: I got this: 
```
drwxr-xr-x    2 root    root                        0 Apr  6  2011 /boot
lrwxrwxrwx    1 root    root                       77 Apr  6  2011 /boot/fitImage -> fitImage--4.9.201+git0+c68230c5b4-r0-jetson-tx2-devkit-cot-20210329095328.bin
-rw-r--r--    1 root    root                 39309146 Apr  6  2011 /boot/fitImage--4.9.201+git0+c68230c5b4-r0-jetson-tx2-devkit-cot-20210329095328.bin
```

Comment (madisongh) [reply layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:9]: You're at least getting a fully-versioned name... is that not enough? That would at least provide a way to have multiple images installed in parallel, if needed.

You can look at the do_deploy task for the fitImage to get the naming scheme used there, and maybe adjust it so the canonical name matches, or is closer to, the naming scheme you're looking for. Just be careful about making any changes to the naming scheme.

File: layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+do_install() {
+    fitImgPath=$(readlink -f ${DEPLOY_DIR_IMAGE}/fitImage)
+    fitImgName=$(basename "$fitImgPath")
+    install -d ${D}/boot
+    install -m 0644 $fitImgPath ${D}/boot/
+    ln -sf $fitImgName ${D}/boot/fitImage
+}
+
+ALLOW_EMPTY_${PN} = "1"
```
Comment (madisongh) [inline layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:15]: Is it really OK for the package to be empty?

File: layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+do_install[depends] += "linux-tegra:do_deploy"
+FILES_${PN} = "/boot/fitImage*"
+PROVIDES = "${PN}"
```
Comment (madisongh) [inline layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:19]: You shouldn't have to set PROVIDES here - '${PN}' is the default for every recipe.

**Review by madisongh — CHANGES_REQUESTED** [review]
Just a couple of minor things to clean up, otherwise looks good.

File: layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb (hunk 2 above)
Comment (ichergui) [reply layers/meta-tegrademo/recipes-bsp/tegra-fitimage/tegra-fitimage_1.0.bb:15]: Ooops!!! No, I should not allow this

**Review by madisongh — APPROVED** [review]

**Pull request merged**